File: store-microservice/src/main/java/com/wbt/storemicroservice/handler/ProductCreatedEventHandler.java
package com.wbt.storemicroservice.handler;

import com.wbt.corelibrary.ProductCreatedEvent;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.kafka.annotation.KafkaHandler;
import org.springframework.kafka.annotation.KafkaListener;
import org.springframework.stereotype.Component;

@Component
@KafkaListener(topics = {"product-created-events-topic"})
public class ProductCreatedEventHandler {

    private final static Logger LOGGER = LoggerFactory.getLogger(ProductCreatedEventHandler.class);

    @KafkaHandler
    public void handler(final ProductCreatedEvent productCreatedEvent) {
        LOGGER.info(">>>> New product created " + productCreatedEvent);
    }

}

File: notification-microservice/src/main/kotlin/com/wbt/notificationmicroservice/handler/ProductCreatedEventHandler.kt
package com.wbt.notificationmicroservice.handler

import com.wbt.corelibrary.ProductCreatedEvent
import org.slf4j.LoggerFactory
import org.springframework.kafka.annotation.KafkaHandler
import org.springframework.kafka.annotation.KafkaListener
import org.springframework.stereotype.Component

@Component
@KafkaListener(topics = ["product-created-events-topic"])
class ProductCreatedEventHandler {

    private val logger = LoggerFactory.getLogger(ProductCreatedEventHandler::class.java)

    @KafkaHandler
    fun handler(productCreatedEvent: ProductCreatedEvent) {
        logger.info("***** Created product message arrived, ${productCreatedEvent.title}")
    }
}
